pagopa/io-app
# fix: [IOBP-798,IOBP-824] Show back button and rename tracking events for no payments method available screen

## Short description
This PR enables the possibility of going back when the uses is in the no payment methods available screen during the payment flow start and also this PR renames the mixpanel events.

## List of changes proposed in this pull request
- Add the possibility to go back in the outcome screen if the outcome is of type `PAYMENT_METHODS_NOT_AVAILABLE`
- Renamed the previous `PAYMENT_METHOD_ERROR_CONTINUE` -> `PAYMENT_NO_SAVED_METHOD_CONTINUE`
- Renamed the previous `PAYMENT_METHOD_ERROR_EXIT` -> `PAYMENT_NO_SAVED_METHOD_EXIT`

## How to test
- Delete every payment method onboarded from your wallet;
- Start a payment flow from the payments tab;
- You should be able to see the no payment method available screen and also the back button

## Preview
https://github.com/user-attachments/assets/af83eb30-7672-4142-9f71-334e0a004052
